src/components/side_panel/pivot/pivot_side_panel/pivot_side_panel_store.ts

```diff
@@ -141,10 +141,19 @@ export class PivotSidePanelStore extends SpreadsheetStore {
         pivot: this.draft,
       });
       this.draft = null;
+      const visiblePivotFormulaIds = new Set<UID>();
+      for (const position of this.getters.getVisibleCellPositions()) {
+        const pivot = this.getters.getPivotIdFromPosition(position);
+        if (pivot) {
+          visiblePivotFormulaIds.add(this.getters.getPivotFormulaId(pivot));
```

Comment:
you can move all this to `isStaticPivotInViewport`. There's a clear overlap between the two pieces of  code and neither of them makes sense without the other.